2015-Fall/kvkunkel/Calculator/Calculator/ViewController.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  ViewController.swift
+//  Calculator
+//
+//  Created by Kristoffer Kunkel on 31/08/15.
+//  Copyright (c) 2015 Kristoffer Kunkel. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController
+{
+    
+    @IBOutlet weak var display: UILabel!
+    
+    @IBOutlet weak var historyDisplay: UILabel!
+
+    var typingNumber: Bool = false
+    var currentFloat: Bool = false
+    
+    var brain = CalculatorBrain()
+    
+    override func viewDidLoad() {
+        historyDisplay.text = ""
+    }
+    
+    @IBAction func appendDigit(sender: UIButton) {
+        println("history: \(brain.getHistory())")
+        let digit = sender.currentTitle!
+        
+        if currentFloat == false {
+            if typingNumber {
+                display.text = display.text! + digit
+            } else {
+                display.text = digit
+                typingNumber = true
+            }
+        } else {
+            if digit != "." {
+                if typingNumber  {
```

Comment:
A 3 level deep if is hard to understand. For a start, the 2nd level starting with `if digit != "." {` could be moved out into a `else if ...`. When that is done it also becomes clear that the two 1st level `if`s contain the same logic. So I think you can simplify this.